src/analysis/convex/basic.lean

```diff
@@ -254,6 +255,71 @@ begin
   simp [set.mem_image, mem_add, eq_comm]
 end
 
+lemma convex.combo_to_vadd {a b : ℝ} {x y : E} (h : a + b = 1) :
+  a • x + b • y = b • (y - x) + x  :=
+  eq.symm (calc
+    b • (y - x) + x = b • (y - x) + (1 : ℝ) • x          : by rw [one_smul]
+                ... = b • (y - x) + (a + b) • x           : by rw [h]
+                ... = (b • y - b • x) + (a • x + b • x)   : by rw [add_smul, smul_sub]
+                ... = a • x + b • y                       : by abel )
+
+/--
+Applying an affine map to an affine combination of two points yields
+an affine combination of the images.
+ -/
+lemma convex.combo_affine_apply {a b : ℝ} {x y : E} {f : affine_map ℝ E E F F}
+  (h : a + b = 1) : f (a • x + b • y) = a • (f x) + b • (f y) :=
+begin
+  rw [convex.combo_to_vadd h],
```

Comment:
If you rewrite twice or `simp only`, then you'll have `f (b • (y - x) + x) = b • (f y - f x) + f x`. Then you can rewrite on `[←vsub_eq_sub, ←f.linear_map_vsub]`, then your goal will be defeq [f.affine_apply_line_map](https://leanprover-community.github.io/mathlib_docs/linear_algebra/affine_space.html#affine_map.affine_apply_line_map).

Reply:
I still need a small calc block to get that `[←vsub_eq_sub]` rewrite to happen at the right place, but otherwise it works.